lighthouse-cli/cli-flags.js

```diff
@@ -12,12 +12,21 @@ const pkg = require('../package.json');
 const printer = require('./printer');
 
 /**
- * @param {string=} manualArgv
+ * @param {(string | string[])[]} arr
+ * @return {string[]}
+ */
+function flatten(arr) {
```

Comment:
fyi a generic "flatten" is impossible in ts :)

Reply:
is it really? how come :)

Reply:
https://www.reddit.com/r/typescript/comments/apfkxh/how_should_i_correctly_type_an_array_flatten/eg82tcu/

I tried to do the multiple variations approach, via jsdoc's `@variation`, but no luck. Unsure if that's fully supported in TS's jsdoc interop.

Reply:
oh like an infinite depth flatten, yeah gotcha :)